### ISSUE TITLE
Concepts/mappings appearing twice in collection

### ISSUE THREAD

**Issue by jamlung-ri** [OpenConceptLab/ocl_issues#1320]
Unsure why, but there is a collection that looks normal except all (or nearly all) concepts are duplicated. This is occurring in the TermBrowser and also in the API results.

Example collection with doubled concepts:
https://app.openconceptlab.org/#/orgs/OHRITechGroup/collections/OHRIF/
https://api.openconceptlab.org/orgs/OHRITechGroup/collections/OHRIF/concepts/

This appears to be happening with the mappings too.

**Comment by jamlung-ri** [comment]
@snyaggarwal Any updates on this? This could be a pretty bothersome bug if it's in Production

**Comment by snyaggarwal** [comment]
@jamlung-ri After a lot of investigation, I figured out the bug. Its because of **self-mappings**, for example:
1. Look at this concept [167281](https://app.openconceptlab.org/#/orgs/CIEL/sources/CIEL/concepts/167281/)
2. This has mapping to self (in the associations section)
3. In the [collection](https://app.openconceptlab.org/#/orgs/OHRITechGroup/collections/OHRIF/) it is [referenced](https://api.openconceptlab.org/orgs/OHRITechGroup/collections/OHRIF/references/812353/) with cascade parameters
4. You can further check [$cascade operation on concept 167281](https://api.openconceptlab.org/orgs/CIEL/sources/CIEL/concepts/167281/$cascade/?cascadeLevels=1)

Below is the solution that I am adding:
1. While $cascade operation is doing its job right, the reference's concepts/mappings may return duplicate results (multiple versions of same concepts/mappings)
2. I have added a fix to dedupe concepts/mappings after references are evaluated and expansion parameters are applied.
2. I deployed the fix on production and generated a [new expansion](https://app.openconceptlab.org/#/orgs/OHRITechGroup/collections/OHRIF/HEAD/expansions/test/), and there are no duplicates now (I have made it as default expansion for now)

**Comment by jamlung-ri** [comment]
That's it! Now the collections are working properly. @paynejd You might want to see this solution. The only loose end might be the Associations section still displaying concepts twice when it's a self-mapping. Otherwise, this is working in QA and in Staging.

![image](https://user-images.githubusercontent.com/55999682/175529642-b70bc67b-1751-4953-98b2-6a6e4fca0d66.png)

**Comment by snyaggarwal** [comment]
@jamlung-ri Can we close this for now and create a new issue for UI behaviour if changes needed there?

**Comment by paynejd** [comment]
Ideally, the TermBrowser would identify a self-map and display it only once
with a different icon. Instead of "-1" for inverse, maybe it would use an
infinity symbol "∞", a cricular arrow "⟳" or something similar? We would
definitely want to add a tooltip on that symbol to indicate that it is a
self-map.

**Comment by snyaggarwal** [comment]
@paynejd Added `∞` for self-mappings with a tooltip and deployed on QA.
Moving this to in testing with another duplicate ticket https://github.com/OpenConceptLab/ocl_issues/issues/1349
